File: Dynamic Programming/acmicpc.net/11726/Main.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileInputStream;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

class Main {
    static final String inputFilePath = "./Dynamic Programming/acmicpc.net/11726/sample.txt";
    // static final int mod = 1000000000;
    // static final String inputFilePath = "./# format/sample.txt";

    public static void main(String[] args) throws Exception {
        System.setIn(new FileInputStream(inputFilePath));
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int N = Integer.parseInt(br.readLine());
        int d[] = new int[N + 1];
        d[0] = 1;
        d[1] = 1;
        for (int i = 2; i < d.length; i++) {
            d[i] = (d[i - 1] + d[i - 2]) % 10007;
        }
        System.out.println(d[N]);
    }
}

File: Dynamic Programming/acmicpc.net/11726/Main.py
# import sys
# input = sys.stdin.readline

file = open("./01 ing/acmicpc.net/sample.txt", "r")
input = file.readline
N = int(input())

d = [0] * (N+1);
d[0]  = 1;
d[1]  = 1;
for i in range(2, N+1):
    d[i] = (d[i-1] + d[i-2]) % 10007;

print(d[N])